src/stores/RoomListStore.js

```diff
@@ -263,6 +264,30 @@ class RoomListStore extends Store {
     }
 
     _recentsComparator(roomA, roomB) {
+        const pinUnread = SettingsStore.getValue("pinUnreadRooms");
```

Comment:
I wonder if implementing the 3 ways to sort in 3 different methods, and then choosing a comparator based on the UserSettings on [line 224](https://github.com/matrix-org/matrix-react-sdk/pull/1936/files#diff-fbd69d95e670ba89b7a7e1abf60c2480L224) wouldn't look nicer.

Reply:
Indeed, given the other comment.